# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/Sandbox.cpp
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 js/xpconnect/src/Sandbox.cpp
--- a/js/xpconnect/src/Sandbox.cpp	Tue Aug 19 16:57:59 2025 +0000
+++ b/js/xpconnect/src/Sandbox.cpp	Tue Aug 19 17:01:50 2025 +0000
@@ -1299,6 +1299,19 @@
     realmOptions.behaviors().setClampAndJitterTime(false);
   }
 
+  if (obj) {
+    nsGlobalWindowInner* window =
+        WindowOrNull(js::UncheckedUnwrap(obj->GetGlobalJSObject(), false));
+    if (window) {
+      const nsCString& localeOverride =
+          window->GetBrowsingContext()->Top()->GetLanguageOverride();
+      if (!localeOverride.IsEmpty()) {
+        realmOptions.behaviors().setLocaleOverride(
+            PromiseFlatCString(localeOverride).get());
+      }
+    }
+  }
+
   const JSClass* clasp = &SandboxClass;
 
   RootedObject sandbox(